core/controlplane/config/templates/cloud-config-controller

```diff
@@ -1602,6 +1634,9 @@ write_files:
                   requests:
                     cpu: 150m
                     memory: 20Mi
+                volumeMounts:
+                - name: kube-dns-config
+                  mountPath: /etc/k8s/dns/dnsmasq-nanny
```

Comment:
Nit but it seems like we're missing the `-configDir` flag which corresponds to this mount for the dnsmasq container. It exists in the [upstream](https://github.com/kubernetes/kubernetes/blob/b09b20b/cluster/addons/dns/kubedns-controller.yaml.in#L122)